Libraries/Components/Touchable/TouchableOpacity.js

```diff
@@ -67,16 +67,24 @@ var TouchableOpacity = React.createClass({
   },
 
   getInitialState: function() {
+    var childStyle = flattenStyle(this.props.style) || {};
+    var opacity = childStyle.opacity === undefined ? 1 : childStyle.opacity;
     return {
       ...this.touchableGetInitialState(),
-      anim: new Animated.Value(1),
+      anim: new Animated.Value(opacity),
     };
   },
 
   componentDidMount: function() {
     ensurePositiveDelayProps(this.props);
   },
 
+  componentDidUpdate: function() {
+    if (!this._hideTimeout && !this._pressIn) {
+      this._opacityInactive();
+    }
+  },
```

Comment:
This is needed because one edge case that I stumble upon. When you are in the "pressing" state if component update it will change opacity, so to prevent that we need to ignore that update and update only when press is released

not sure if I am clear, if you need more info I will try to explain that better